service-base/src/main/java/org/eclipse/hono/service/credentials/BaseCredentialsService.java

```diff
@@ -109,31 +109,61 @@ protected void doStop(final Future<Void> stopFuture) {
-                String type = payload.getString(FIELD_TYPE);
-                String authId = payload.getString(FIELD_AUTH_ID);
-                log.debug("getting credentials [{}:{}] of tenant [{}]", type, authId, tenantId);
-                getCredentials(tenantId, type, authId, result -> reply(regMsg, result));
-            }
+        final JsonObject body = regMsg.body();
+        if (checkMessagePartsForNull(body,"body",regMsg)) {
+            return;
+        }
+        final String tenantId = body.getString(MessageHelper.APP_PROPERTY_TENANT_ID);
+        if (checkMessagePartsForNull(tenantId,MessageHelper.APP_PROPERTY_TENANT_ID,regMsg)) {
+            return;
+        }
+        final String subject = body.getString(MessageHelper.SYS_PROPERTY_SUBJECT);
+        if (checkMessagePartsForNull(subject,MessageHelper.SYS_PROPERTY_SUBJECT,regMsg)) {
+            return;
+        }
+
+        final JsonObject payload = getRequestPayload(body,regMsg);
+
+        switch (subject) {
+            case OPERATION_GET:
+                if (payload == null) {
+                    // payload did not exist or it was not of type json
+                    log.info("get credentials request with non json payload not supported");
+                    reply(regMsg, CredentialsResult.from(HTTP_BAD_REQUEST));
+                    return;
+                }
+                processCredentialsMessageGetOperation(regMsg, tenantId, payload);
                 break;
             default:
                 log.info("operation [{}] not supported", subject);
                 reply(regMsg, CredentialsResult.from(HTTP_BAD_REQUEST));
-            }
-        } catch (ClassCastException e) {
-            log.debug("malformed request message");
-            reply(regMsg, CredentialsResult.from(HTTP_BAD_REQUEST));
-        } catch (NullPointerException ne) {
-            log.debug("mandatory parameters in message were null");
+        }
+    }
+
+    private boolean checkMessagePartsForNull(final Object msgPart, final String msgPartName,final Message<JsonObject> regMsg) {
+        if (msgPart == null) {
+            log.info("credentials request with {} 'null' not supported",msgPartName);
             reply(regMsg, CredentialsResult.from(HTTP_BAD_REQUEST));
+            return true;
         }
+        return false;
+    }
+
+    private void processCredentialsMessageGetOperation(final Message<JsonObject> regMsg, final String tenantId, final JsonObject payload) {
+        if (checkMessagePartsForNull(payload,"payload",regMsg)) {
+            return;
+        }
+        final String type = payload.getString(FIELD_TYPE);
+        if (checkMessagePartsForNull(type,FIELD_TYPE,regMsg)) {
+            return;
+        }
+
+        final String authId = payload.getString(FIELD_AUTH_ID);
+        if (checkMessagePartsForNull(authId,FIELD_AUTH_ID,regMsg)) {
+            return;
+        }
+        log.debug("getting credentials [{}:{}] of tenant [{}]", type, authId, tenantId);
```

Comment:
since we need to check each property individually anyway, we could improve readability a lot by making the checks more explicit, e.g. something like this:
```
final JsonObject payload = getPayload(body);
if (payload == null) {
  reply(..., ...);
} else {
 final String tenantId = body.getString(MessageHelper.APP_PROPERTY_TENANT_ID;
 final String type = payload.getString(FIELD_TYPE);
 final String authId = payload.getString(FIELD_AUTH_ID);
 if (tenantId == null) {
  reply(..., ...);
 } else if (type == null) {
   reply(..., ...);
 } else if (authId == null) {
   reply(..., ...);
 } else {
   log.debug("getting credentials [{}:{}] of tenant [{}]", type, authId, tenantId);
   getCredentials(tenantId, type, authId, result -> reply(regMsg, result));
 }
}

The `checkMessagePartsForNull` operation does not really save much code but its usage is kind of confusing IMHO
WDYT?